src/main/java/TTT/databaseUtils/CustomUserDAO.java

```diff
@@ -16,22 +16,36 @@ public class CustomUserDAO {
     private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
     private final SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
 
-    public boolean saveUser(CustomUser customUser){
+    public boolean saveUser(CustomUser customUser) {
+
+        if (findCustomUser(customUser.getEmail()) != null) {
+            return false;
+        }
+        if (customUser.getPassword().length() < 7) {
+            return false;
+        }
+
+        customUser.setCustomUserName("yourName");
+
+        Session session = null;
+        Transaction transaction = null;
         try {
```

Comment:
session can be created in a try with resources

Reply:
ok! Done.